### ISSUE TITLE
Switch to CMake and fix compilation with VS2022

### ISSUE THREAD

**Issue by mrexodia** [x64dbg/x64dbg#3412]
This has been on my personal roadmap for a while, but wanted to add an issue to track it.

- [x] Add a CMakeLists.txt in parallel to x64dbg.sln
- [x] Support VS2022:
   - Fix all warnings
   - Fix Visual Studio analyzer results
- [x] Add TitanEngine/GleeBug as submodules
- [x] Add an option to automatically download a supported Qt5 LTS version (VS2019)
- [x] Integrate clang-tidy
- [x] Clean up the project root
- [ ] Support “offline” builds (ready-to-compile source package)
- [ ] Create an external plugin for Scylla (latest fork)
- [x] Upgrade dbghelp.dll and msdia140.dll to the latest version
- [x] Restore delayed imports
- [x] Restore signing functionality for latest MSVCRT
- [x] Remove TitanEngine/DeviceNameResolver and other internal libraries from the pluginsdk
- [x] Update compilation instructions in the wiki
- [ ] Update the [QtPlugin](https://github.com/x64dbg/QtPlugin) template and PluginTemplate
- [x] Delete the old `.vcxproj` and `x64dbg.pro` files
- [x] Fix the relative submodule path when forked (`url = ../../x64dbg/deps`)
- [ ] Make sure the `exe` project is the default startup project in the generated solution
- [x] Update CONTRIBUTING.md to match the new workflow
- [x] Remove licenses from `docs` folder and pull them dynamically during the build to avoid multiple licenses being detected for the repository

**Comment by torusrxxx** [comment]
Next milestone of adding AVX-512 support as part of #3136, will require modification to debug engines.

**Comment by mrexodia** [comment]
Do we actually need to support AVX-512? They removed it from the new CPUs and my CPU from 2020 doesn't even have support for it 😅

**Comment by torusrxxx** [comment]
It's just the incompetency of Intel. AMD has no problem supporting AVX-512. I will test it on a computer with AVX-512 myself.

**Comment by mrfearless** [comment]
I was looking at Steam Hardware Survey just yesterday when Steam prompted me to take my info, and I had a look at the last section with the cpu features. For myself I was curious to see the uptake of SSE etc, but noticed the low usage of AVX512 currently.

Of course the sample size is limited to gamers, but still interesting to see what is current, and what may be more prevalent in future.

https://store.steampowered.com/hwsurvey/

**Comment by mrexodia** [comment]
Yeah the prevalence might be low, but supporting it transparently shouldn’t be _that_ much of a burden. So why not? 🤷‍♂️

**Comment by mrexodia** [comment]
I merged the CMake project as an initial step, to hopefully at least make it easier for new people to contribute. It will likely be unavoidable to break the Qt GUI plugins when upgrading the toolchains (name mangling is different), so I think we can upgrade to the latest Qt that has XP compatibility.

We would then start informing users about full XP deprecation (perhaps we can provide instructions to use something like https://github.com/Chuyu-Team/YY-Thunks or https://github.com/Skulltrail192/One-Core-Api-Source). XP has been out of support since 2014, which is more than 10 years, but I feel it keeps new contributors out and the maintenance burden is constant.
